consensus/consensus_leader.go

```diff
@@ -31,13 +31,24 @@ var (
 
 // WaitForNewBlock waits for the next new block to run consensus on
 func (consensus *Consensus) WaitForNewBlock(blockChannel chan *types.Block, stopChan chan struct{}, stoppedChan chan struct{}, startChannel chan struct{}) {
+	// gensis block is the first block to be processed.
+	// But we shouldn't start consensus yet, as we need to wait for all validators
+	// received the leader's pub key which will be propogated via Pong message.
+	// After we started the first consensus, we will go back to normal case to wait
+	// for new blocks.
+	// The signal to start the first consensus right now is the sending of Pong message (SendPongMessage function in node/node_handler.go
+	// but it can be changed to other conditions later
+	first := true
 	go func() {
```

Comment:
[Nit] This background function is the entirety of this function; can we convert this to foreground and instead call:
```
go s.consensus.WaitForNewBlock(s.blockChannel, s.stopChan, s.stoppedChan)
```
from `api/service/consensus/service.go`?  Two reasons:

1. Less indentation.
2. The function name sounds like a synchronous, blocking function but it is asynchronous/background.

Reply:
this is the old code and I don't feel I should touch it in my PR.  My PR is focusing on getting consensus working with libp2p.